pkg/pwgen/pwrules/gen.go

```diff
@@ -99,12 +103,51 @@ func fetchRules() (map[string]jsonRule, error) {
 		return nil, err
 	}
 	var jr map[string]jsonRule
-	if err := json.NewDecoder(resp.Body).Decode(&jr); err != nil {
+	if err := json.NewDecoder(&cleaningReader{src: resp.Body}).Decode(&jr); err != nil {
 		return nil, err
 	}
 	return jr, nil
 }
 
+type cleaningReader struct {
+	src io.Reader
+	rdr io.Reader
+}
+
+func (c *cleaningReader) init() error {
+	if c.rdr != nil {
+		return nil
+	}
+
+	var buf bytes.Buffer
+	scanner := bufio.NewScanner(c.src)
+	for scanner.Scan() {
+		line := scanner.Text()
+		skip := false
+		// skip two broken entries
+		for _, needle := range []string{"fidelity.com", "hkexpress.com"} {
+			if strings.Contains(line, "\""+needle+"\"") {
+				skip = true
+			}
+		}
+		if skip {
+			scanner.Scan()
```

Comment:
Are these two Scan really meant to be here?

Reply:
Yes, I should add a comment.

This is my very naive way of working around seriously broken JSON.

It looks for the first line of a three line struct and skips the two other lines. Naive and maybe stupid, but effective.